config/routes.rb

```diff
@@ -36,6 +36,18 @@
           resources :search_references, only: [:show, :index, :destroy, :create, :update]
         end
       end
+
+      resources :chemicals, only: %i[index create] do
+        get     '',                                       to: 'chemicals#show'
+        patch   '(:chemical_name_id/:new_chemical_name)', to: 'chemicals#update'
+        put     '(:chemical_name_id/:new_chemical_name)', to: 'chemicals#update'
+       #delete  '',                                       to: 'chemicals#delete'
+        get     'map',                                    to: 'chemicals#show_map'
+        post    'map/:goods_nomenclature_sid',            to: 'chemicals#create_map'
```

Comment:
@mcumcu why do we use both `goods_nomenclature_sid` and `goods_nomenclature_item_id ` ?

Reply:
I've refactored the routes. The new ones do use both item_id and sid. Here's why:
- `post    'map/:goods_nomenclature_item_id'` - The endpoint could be changed to accept the goods_nomenclature_sid instead, and better conform to the Rails/API routing convention of using the object's primary key as the identifier, but on the Admin UI, the user would have to:
  1.  enter the goods_nomenclature_sid in a text field, and/or
  2.  select the GN from a drop-down select field, or
  3.  could also implement a JS lookup-as-you-type feature on the text field or select field
- `patch|put   'map/:goods_nomenclature_sid(/:goods_nomenclature_item_id)'` - sid is used for the existing mapping that is to be updated, the new commodity to map to is identified by item_id, since that's what the user enters in the admin UI (see above for other options)